Tasks/RunDistributedTests/CheckTestAgentCompat.ps1

```diff
@@ -0,0 +1,76 @@
+function Check-AgentCompat($ProductVersion)
+{
+    try
+    {
+        $InstalledVersionRegValueName = "Version"
+        $InstalledVersionRegKey = ("SOFTWARE\Microsoft\DevDiv\vstf\Servicing\{0}\testExecCore" -f $ProductVersion)
+
+        if ($InstalledVersionRegKey -and $InstalledVersionRegValueName)
+	    {
+          $versionString = $null
+
+          #if 64bit OS, check 64bit registry view first
+          if ((Get-WmiObject -Class Win32_OperatingSystem -ComputerName "localhost" -ea 0).OSArchitecture -eq '64-bit')
+          {
+            $versionString = Get-RegistryValue -RegistryHive LocalMachine -Key $InstalledVersionRegKey -Value $InstalledVersionRegValueName -RegistryView Registry64
+          }
+          
+          if($versionString -eq $null)
+          {          
+            $versionString = Get-RegistryValue -RegistryHive LocalMachine -Key $InstalledVersionRegKey -Value $InstalledVersionRegValueName -RegistryView Registry32
+          }
+
+          if($versionString)
+          {
+            $versionArray = $versionString.split('.')            
+
+            $minorVersion=$null
+            $majorVersion=$null
+            [Int]::TryParse($versionArray[0], [ref]$majorVersion)
+            [Int]::TryParse($versionArray[2], [ref]$minorVersion)
+
+            if($majorVersion -eq 14 -and $minorVersion -le 24712)
+            {
+              # Comparing with version of Agent last shipped that does not contain the changes.
+              Write-Error "Update Test Agent to be able to run tests from Test plan." -Verbose
```

Comment:
can we just Write-Verbose this and then based on this msg print localized form of this string. reason is because error msgs shuld be localized and we still don't have the story for localization of remote scripts